src/Fixer/Whitespace/MethodChainingIndentationFixer.php

```diff
@@ -0,0 +1,172 @@
+                if ($this->needLineBreak($index - 1, $tokens)) {
+                    $tokens[$index - 1]->setContent($tokens[$index - 1]->getContent().$lineEnding);
+                    --$index;
+                    continue;
+                }
+
+                $prev = $tokens[$index - 1];
+                $currentWhitespaces = $this->getLineBreak($prev->getContent());
+
+                if (null !== $currentWhitespaces) {
+                    $prevMeaningIndex = $tokens->getPrevMeaningfulToken($index);
+                    $rightWhitespaces = $this->getRightIndents($prevMeaningIndex, $tokens);
+
+                    if ($currentWhitespaces !== $rightWhitespaces) {
+                        $prev->setContent($lineEnding.$rightWhitespaces);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return string
+     */
+    private function getRightIndents($index, Tokens $tokens)
+    {
+        $indent = $this->whitespacesConfig->getIndent();
+
+        for ($i = $index; $i >= 0; --$i) {
+            if ($i > 0) {
+                $codeToFindIndents = $tokens->generatePartialCode($i - 1, $i);
+            } else {
+                $codeToFindIndents = $tokens[$i]->getContent();
+            }
+
+            $currentWhitespaces = $this->getLineBreak($codeToFindIndents);
+
+            if (null !== $currentWhitespaces) {
+                if ($tokens[$i + 1]->isGivenKind(T_OBJECT_OPERATOR) || $this->isMultiLineMethod($i, $index, $tokens)) {
+                    return $currentWhitespaces;
+                }
+
+                return $currentWhitespaces.$indent;
+            }
+        }
+
+        return $indent;
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    private function needLineBreak($index, Tokens $tokens)
+    {
+        $prevMeaningful = $tokens->getPrevMeaningfulToken($index);
+        $isComment = false;
+
+        for ($i = $index; $i > $prevMeaningful; --$i) {
+            if ($tokens[$i]->isGivenKind(T_OBJECT_OPERATOR) || null !== $this->getLineBreak($tokens[$i]->getContent())) {
+                return $isComment;
+            }
+
+            if ($tokens[$i]->isComment()) {
+                $isComment = true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $content
+     *
+     * @return string|null
+     */
+    private function getLineBreak($content)
+    {
+        if (preg_match('/\R{1}([ \t]*)$/', $content, $matches)) {
+            return $matches[1];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param int    $start
+     * @param int    $end
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    private function isMultiLineMethod($start, $end, Tokens $tokens)
+    {
+        if (')' === $tokens[$end]->getContent()) {
```

Comment:
@keradus So, what the right way to do that?
I can't use equals here, because sometimes token is not an array
![selection_007](https://cloud.githubusercontent.com/assets/1892918/26702823/69a8624a-472f-11e7-9f5c-a4887c2bf7b5.png)
But sometimes is 
![selection_008](https://cloud.githubusercontent.com/assets/1892918/26702831/74d41326-472f-11e7-9c20-a35b2738c010.png)
It is because of **BraceClassInstantiationTransformer**